src/components/hooks/useMeeting.js

```diff
@@ -1,28 +1,50 @@
 import {useEffect, useContext, useState} from 'react';
+import {from} from 'rxjs';
+import {delayWhen, flatMap, take} from 'rxjs/operators';
 
 import {AdapterContext} from '../../components/';
 
 /**
  * Custom hook that returns meeting data of the given ID.
+ * If no ID is given, a `meetingDestination` must be provided to
+ * create a new meeting.
  *
- * @param {string} meetingID  ID of the meeting for which to return data.
+ * @param {string} meetingID           ID of the meeting for which to return data.
+ * @param {string} meetingDestination  Virtual location where the meeting should take place
  * @returns {Meeting} Data of the meeting
  */
-export default function useMeeting(meetingID) {
+export default function useMeeting(meetingID, meetingDestination) {
   const [meeting, setMeeting] = useState({});
   const {meetingsAdapter} = useContext(AdapterContext);
 
   useEffect(() => {
+    let subscription;
     const onError = (error) => {
       throw error;
     };
-    const subscription = meetingsAdapter.getMeeting(meetingID).subscribe(setMeeting, onError);
+
+    if (meetingID) {
+      subscription = meetingsAdapter.getMeeting(meetingID).subscribe(setMeeting, onError);
+    } else if (meetingDestination) {
+      // Create a meeting, start event listeners by subscribing to getMeeting,
+      // wait for local media to get added and complete
+      subscription = meetingsAdapter
+        .createMeeting(meetingDestination)
+        .pipe(
```

Comment:
It would be nice if the `createMeeting` subscription could handle all this logic for us instead of needing it in the hook.

Reply:
that could be another action item for enhancement maybe?

Reply:
Agreed, let's do it in a separate PR. However, not sure that I like the idea of `createMeeting` code calling `getMeeting` behind the scenes

Reply:
we could either wrap all methods into one or figure out if there are any other cases when we need to invoke `createMeeting` by itself? if not, then we can move all the logic inside `createMeeting`